framework/framework.py

```diff
@@ -166,6 +167,20 @@ def handle_offer(self, driver, offer):
         _log.debug("All tasks are running")
         return None
 
+    def get_role(self, offer):
+        """
+        Helper method to disect an offer and determine which role the resources
+        belong to.
+        :param offer:
+        :return: A string representation of the role. This can be "*" or a string.
+        """
+        # If we find anything besides "*", use it as the role. Otherwise, just use "*"
+        role = "*"
```

Comment:
We should add some details as to why we can do this simplified processing:
-  Basically an offer only contains non-zero resources for either Public or Non-public (*)